### DISCUSSION TITLE
Stops working after a few weeks

### DISCUSSION THREAD

**Discussion by thrangster** [homebridge-simplisafe3/homebridge-simplisafe3#285] [Q&A] [answered]
Hi - great plug in, but I am having an issue where after a few weeks, the plug-in simply stops working (all accessories disappear in Homebridge and the Home app...)

I am performing the authentication process as instructed, pasting in the resultant URL - and everything works quite well. But then some weeks later it stops. I need to re-authenticate and it starts working again - until the same things happens again

Is this a reality of the new Simplisafe authentication process, or do certain conditions trigger this problem?

Thanks

**Comment by shamoon** [comment, marked as answer]
Hi @thrangster If you enable `debug` and check the logs it might be helpful in figuring out why authentication fails. The plug-in is designed to obtain a new token every ~hour to prevent this but yes, it will happen. Oddly it’s been extremely rare for me compared to others, only once in about 4 months.

Anyway, something I implemented recently based on another users’ request is a way to get notified of authentication failures which you might find helpful. Please see: https://github.com/homebridge-simplisafe3/homebridge-simplisafe3#authentication-failure-notifications

**Reply by thrangster** [reply]
Yes, I enabled debug this morning - how/where do I check to provide feedback when it does occur?

Even the Simplsafe apps requires you to log in again after 30 days, so I wonder if the act of doing that disrupts other authenticated devices?

I will look into the link you posted... I use it to automate alarms and locks, so important for me to at least know what's going on...

**Reply by thrangster** [reply]
So I installed the homebridge messenger plug in, but don'd understand how to configure it... I look at the documentation, but half of it is foreign to me.... if you or anyone knows how to set this up to be notified when the authentication for the Simplisafe plan in flats, that would be great... thanks

**Reply by shamoon** [reply]
> Yes, I enabled debug this morning - how/where do I check to provide feedback when it does occur?

You can post here if it looks like it might be revealing.

> Even the Simplsafe apps requires you to log in again after 30 days, so I wonder if the act of doing that disrupts other authenticated devices?

Logging into the app shouldn't affect this plug-in, and it shouldn't be limited to 30 days.

> So I installed the homebridge messenger plug in, but don'd understand how to configure it... I look at the documentation, but half of it is foreign to me.... if you or anyone knows how to set this up to be notified when the authentication for the Simplisafe plan in flats, that would be great... thanks

Yea Id look at the documentation, there are a lot of options for types of notifications you can setup, email, etc. There are also other types of notification plugins, its really up to you. Once you have a notification method setup (in whatever plugin you choose) you would setup an automation on the `Status Fault` property of the alarm. I used the (free) Eve app for the automation.

Here's my messenger config:
```
        {
            "name": "Messenger",
            "services": {
                "pushover": {
                    "user": "***",
                    "token": "***"
                },
                "email": {
                    "recipient": "email@domain.com",
                    "smtpServer": "smtp.gmail.com",
                    "smtpPort": 587,
                    "smtpSecure": true,
                    "smtpUsername": "email@domain.com",
                    "smtpPassword": "***"
                }
            },
            "messages": [
                {
                    "name": "SS Auth Pushover",
                    "type": "pushover",
                    "text": "SimpliSafe authentication failure detected",
                    "priority": 1,
                    "sound": "pushover"
                },
                {
                    "name": "SS Auth Email",
                    "type": "email",
                    "text": "SimpliSafe authentication failure detected",
                    "recipients": "email@domain.com",
                    "priority": 0,
                    "sound": "pushover"
                }
            ],
            "accessory": "HomebridgeMessenger"
        }
```

Here's a screenshot of my automation in the Eve app and the Home.app. Note that the Home app just says Simplisafe "Changes" because in the Home app you cant setup the trigger based on `Status Fault`
| Eve App | Home App |
| ------------- | ------------- |
| ![IMG_7115](https://user-images.githubusercontent.com/4887959/150391789-6ec2813a-ef40-4324-a151-db7320405d94.jpeg) | ![IMG_7116](https://user-images.githubusercontent.com/4887959/150391797-324d3514-bf13-492e-a061-67f61001e8b1.jpeg) |

**Reply by dwells-git** [reply]
> Here's a screenshot of my automation in the Eve app and the Home.app. Note that the Home app just says Simplisafe "Changes" because in the Home app you cant setup the trigger based on `Status Fault`

Hey, question on this: if the automation triggers whenever the SimpliSafe main unit changes, won't that cause it to flip the switch (i.e. send the email) whenever anything happens to the unit, such as being armed or disarmed?

I've currently got homebridge-messenger installed with the switches showing in Homebridge and I manually triggered the test switch and it works for sending the email, I'm trying to make it so I get an email specifically in the event of an authentication failure happening.

**Comment by thrangster** [comment]
thats - I’ll see if my 61 year old brain can figure this out….!

**Comment by thrangster** [comment]
Im a bit stuck on the user and token fields… user can be simply any name?

Also, Messenger documentation says token is “your applications API token” 

I tried entering the MAC address-like looking number in the Home Alarm section on my config, but I still don't see anything in Home.

            "name": "Home Alarm",
            "cameras": true,
            "debug": true,
            "sensorRefresh": 15,
            "persistAccessories": false,
            "resetSimpliSafeId": false,
            "cameraOptions": {
                "enableHwaccelRpi": false
            },
            "_bridge": {
                "username": “xxxxxxxxxx",
                "port”: xxxx
            },
            "platform": "homebridge-simplisafe3.SimpliSafe 3"


I see this in the Accessories tab in Homebridge





These should appear as individual tiles in Home, correct?


Separately, I am seeing tons of these error in the log...

```
[20/01/2022, 16:51:37] [Home Alarm] Sensor refresh received an error from the SimpliSafe API: {
  type: 'SettingsInProgress',
  message: 'The Settings Requested are already being modified and cannot be accessed at this time',
  code: '200',
  statusCode: 409,
  props: {}
}
[20/01/2022, 16:51:52] [Home Alarm] Sensor refresh received an error from the SimpliSafe API: {
  type: 'SettingsInProgress',
  message: 'The Settings Requested are already being modified and cannot be accessed at this time',
  code: '200',
  statusCode: 409,
  props: {}
}
[20/01/2022, 16:52:07] [Home Alarm] Sensor refresh received an error from the SimpliSafe API: {
  type: 'SettingsInProgress',
  message: 'The Settings Requested are already being modified and cannot be accessed at this time',
  code: '200',
  statusCode: 409,
  props: {}
}
[20/01/2022, 16:52:22] [Home Alarm] Sensor refresh received an error from the SimpliSafe API: {
  type: 'SettingsInProgress',
  message: 'The Settings Requested are already being modified and cannot be accessed at this time',
  code: '200',
  statusCode: 409,
  props: {}
}
[20/01/2022, 16:52:37] [Home Alarm] Sensor refresh received an error from the SimpliSafe API: {
  type: 'SettingsInProgress',
  message: 'The Settings Requested are already being modified and cannot be accessed at this time',
  code: '200',
  statusCode: 409,
  props: {}
}
[20/01/2022, 16:52:52] [Home Alarm] Sensor refresh received an error from the SimpliSafe API: {
  type: 'SettingsInProgress',
  message: 'The Settings Requested are already being modified and cannot be accessed at this time',
  code: '200',
  statusCode: 409,
  props: {}

```

And some of these:

```
[20/01/2022, 21:21:17] [Home Alarm] Sensor refresh received an error from the SimpliSafe API: {
  type: 'GatewayTimeout',
  message: 'The message timed out connecting to the base station.',
  code: '009',
  statusCode: 504,
  props: {}
}
[20/01/2022, 21:21:17] [Home Alarm] Sensor refresh received an error from the SimpliSafe API: {
  type: 'GatewayTimeout',
  message: 'The message timed out connecting to the base station.',
  code: '009',
  statusCode: 504,
  props: {}
}
[20/01/2022, 21:21:17] [Home Alarm] Sensor refresh received an error from the SimpliSafe API: {
  type: 'GatewayTimeout',
  message: 'The message timed out connecting to the base station.',
  code: '009',
  statusCode: 504,
  props: {}
```



Thank you very much…

**Reply by shamoon** [reply]
Pushover is a specific service ( https://pushover.net/ ) you don’t necessarily have to use, and you get that info from your account after signing up. You’ll see my config also includes an email notification that uses gmail, etc. Messenger plug-in allows for a few different means of sending notifications, also IFTTT for example

**Reply by thrangster** [reply]
I think the issue is the Messenger actions aren't appearing in Home as the icons I can see or action...I have email set up as well as the second part of the config. I can't find or set up the messaging in Home...

Just to be clear, what do I enter in the "user" and "token" fields in the Messenger config? I think that's where the problem is.... thanks kindly

**Comment by pbr90x** [comment]
fancypantsifier> Ironically, I could not find how to access that Status
Fault from Eve though.

In the Eve iOS app, I was able to setup a rule with a trigger on the
Simplisafe status fault.

Automation > Rules

 + Add Rule


Next


Add Trigger > Other Value >


Types : General Fault


Select your SimpliSafe device (whatever you named the SS alarm in HomeKit),
select Yes

Q: Can someone explain how to accomplish the same w/ "shortcut-based
automations" natively in iOS or macOS, without having to use the 3rd party
Eve iOS app?

thrangster> Im a bit stuck on the user and token fields… user can be simply
any name?

This was confusing for me at first too. In the Messenger plugin's Pushover
settings, set User to the "User Key" you were provided when you created
your Pushover account. Next, in Pushover, you will need to create a new
Application / API Token <https://pushover.net/apps/build>. I named this
application one "Homebridge". Fill in the Messenger plugin's Token field
with the the Pushover API token that you created for this purpose.

**Comment by bee-easy** [comment]
hello - were you able to figure this out?  I am having the same issue.  I did the new authorization and everything was working great until last week where it just disappeared from my Home app.

**Reply by shamoon** [reply]
Authentication failure shouldn’t cause the devices to disappear from your home, just get “No response”. Check the ‘persist accessories’ setting is turned on (by default it is)

**Reply by thrangster** [reply]
> hello - were you able to figure this out? I am having the same issue. I did the new authorization and everything was working great until last week where it just disappeared from my Home app.



> Authentication failure shouldn’t cause the devices to disappear from your home, just get “No response”. Check the ‘persist accessories’ setting is turned on (by default it is)

Intersting - this is off for me, and I never went in there... turned it on.

Still, I can't even figure out how to get messenger two work (I dont see the email tile in Home to configure a notification....)

**Reply by shamoon** [reply]
Yea, configuring another plugin is a little outside the scope of this one. I tried to explain best I can in my comment above https://github.com/homebridge-simplisafe3/homebridge-simplisafe3/discussions/285#discussioncomment-2008529 but if youre still having trouble you may want to try Reddit / Discord or even the messenger plugin's Github. Again, there are also other similar plugins you can use that might be easier to setup.

**Reply by bee-easy** [reply]
my "persist accessories" was unchecked as well and just checked the box and restarted.  this is the error is am seeing "SimpliSafe login failed with error: Error: Failed refreshing token: Error: Request failed with status code 403
    at SimpliSafe3AuthenticationManager.refreshCredentials (/usr/local/lib/node_modules/homebridge-simplisafe3/common/authManager.js:222:13)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)"

**Reply by bee-easy** [reply]
oh geez, this is my fault.. I had to do the reauth again and now it reappeared and working.

**Reply by thrangster** [reply]
> oh geez, this is my fault.. I had to do the reauth again and now it reappeared and working.

hopefully it sticks - the problem some of us our seeing is losing auth every 2-4 weeks...

**Reply by shamoon** [reply]
I appreciate this is frustrating but ultimately the authentication process in the plugin works and does its thing to re-authenticate as expected the vast majority of the time. Im not sure how much I can do to prevent the occasional hiccups, they might just be occasional connectivity issues with bad timing (e.g. when plugin is trying to authenticate). As I have said, if I can see logs (with debug enabled) at the moment authentication craps out it might help pinpoint something I can try to prevent, but ya know, it might not... 

Im trying to keep an eye on it myself as well, of course.

**Reply by bee-easy** [reply]
I will keep an eye on it.. I tend to monitor it weekly to ensure all my plugins are working appropriately.

**Comment by thrangster** [comment]
Does this seem like it should work?

"accessories": [
        {
            "name": "Homebridge Messenger",
            "services": {
                "pushover": {
                    "user": "user code from pushover account",
                    "token": "token from pushover for API I Created"
                },
                "email": {
                    "recipient": "name@gmail.com",
                    "smtpServer": "smtp.gmail.com",
                    "smtpPort": 587,
                    "smtpSecure": true,
                    "smtpUsername": "name@gmail.com",
                    "smtpPassword": "password"
                }
            },
            "messages": [
                {
                    "name": "SS Auth Pushover",
                    "type": "pushover",
                    "text": "Simplisafe Authentication Failure detected",
                    "priority": 1,
                    "sound": "pushover"
                },
                {
                    "name": "SS Auth Email",
                    "type": "email",
                    "text": "Simplisafe Authentication Failure detected",
                    "recipients": "name@gmail.com",
                    "priority": 0,
                    "sound": "pushover"
                }
            ],
            "accessory": "HomebridgeMessenger"

I dont see the tiles in Home....
